cluster_follow.go

```diff
@@ -274,24 +276,28 @@ func (db *DB) processFollowers(stop <-chan interface{}) {
 						}
 					}
 				}
+				earliestOffsetByStream[stream] = earliestOffset
+			}
 
-				stopWALReader := stopWALReaders[stream]
+			for stream, stopWALReader := range stopWALReaders {
 				if stopWALReader != nil {
 					db.log.Debugf("Stopping WAL reader for %v", stream)
 					stopWALReader()
 				}
+			}
 
+			for stream, startStream := range streams {
```

Comment:
It looks like you were previously iterating over `newlyJoinedStreams` here - does it matter that you're now iterating over all `stream`s?

Reply:
Yes, that's actually the heart of the change. Instead of just restarting the stream that recently rejoined, we restart all of them.